paywall/src/middlewares/web3Middleware.js

```diff
@@ -105,6 +106,24 @@ const web3Middleware = config => ({ getState, dispatch }) => {
   return function(next) {
     return function(action) {
       if (action.type === ADD_TRANSACTION) {
+        const {
+          account,
+          router: {
+            location: { pathname, hash },
+          },
+        } = getState()
+        const { lockAddress } = lockRoute(pathname + hash)
+        const accountAddress = account && account.address
+        const { to, from } = action.transaction
+        if (from === accountAddress && to === lockAddress) {
+          // this is probably a key purchase transaction, so let's make sure we know it is in the UI
+          // pending key transactions will never see this detail otherwise
+          action.transaction.type = TRANSACTION_TYPES.KEY_PURCHASE
```

Comment:
Nah, this is not right we should not do guesswork here. Web3Service **has** the ability to retrieve that info (and used to!), so let's make sure it does.